File: 014. Nth Fibonacci/NthFibonacciOnO1.java
import java.util.*;

class NthFibonacciOnO1 {
    public static int getNthFib(int n) {
        int[] base = { 0, 1 };
        int counter = 3;
        while (counter <= n) {
            int nextFib = base[0] + base[1];
            base[0] = base[1];
            base[1] = nextFib;
            counter++;
        }
        return n > 1 ? base[1] : base[0];
    }
}

File: 014. Nth Fibonacci/NthFibonacciOnO1.kt
package com.algoexpert.program

fun getNthFib(n: Int): Int {
    var lastTwo = Pair(0, 1)
    var counter = 3
    while (counter <= n) {
        var nextFib = lastTwo.first + lastTwo.second
        lastTwo = Pair(lastTwo.second, nextFib)
        counter++
    }
    return if (n <= 1) lastTwo.first else lastTwo.second
}
